### DISCUSSION TITLE
Time range filter creates a non-optimal SQL query

### DISCUSSION THREAD

**Discussion by fertek** [apache/superset#24144] [General]
Hello, I have a question about optimizing a SQL query that is built for a chart in **Explore view**.

If I use **time range filter**, I get the following SQL statement:

```
SELECT sum(quantity) AS "quantity"
FROM
  (SELECT "date_column",
          "quantity"
   FROM public.demo_table) AS virtual_table

WHERE "date_column" >= TO_TIMESTAMP('2023-02-19 00:00:00.000000', 'YYYY-MM-DD HH24:MI:SS.US')
  AND "date_column" < TO_TIMESTAMP('2023-05-19 00:00:00.000000', 'YYYY-MM-DD HH24:MI:SS.US');
```

This query takes about **3 minutes** to complete.

But if I modify the WHERE clause (in SQL Lab), I get the same result much faster:

```
SELECT sum(quantity) AS "quantity"
FROM
  (SELECT "date_column",
          "quantity"
   FROM public.demo_table) AS virtual_table

WHERE "date_column" >= '2023-02-19'::date
  AND "date_column" < '2023-05-19'::date;
```

This query takes about **3 seconds** to complete.

Is it possible to configure the chart to work with this faster variant?

Why is the `TO_TIMESTAMP(…)` variant so much slower?

Thanks.

**Comment by ved-kashyap-samsung** [comment]
Your database might have some caching mechanism internally. And in some cases when we use function related to date time like TO_TIMESTAMP caching is not considered. But when you pass the date in as it is then first time runs query but second time it might return result from cache. Please check if this is the case.
